src/main/java/com/example/document/service/DocumentService.java

```diff
@@ -37,11 +36,20 @@ public List<DocumentWithEmployeeDTO> uploadExcelDocument(MultipartFile file) thr
 
         List<DocumentEntry> docEntryList = new ArrayList<>();
 
+//        Employee validation
+        for (int i = 1; i <= sheet.getLastRowNum(); i++) {
+            Row row = sheet.getRow(i);
+            Long employeeId = (long) row.getCell(2).getNumericCellValue();
+            employeeClient.getEmployeeById(employeeId);
+        }
+
```

Comment:
I used loop twice because I wanted to validate all the document entries before, and make condition, if the file or list of document entries contains even one incorrect entry, the process should be suspended

Reply:
Let's combine these loops to reduce amount of http requests